### ISSUE TITLE
Yew html macro error Expect Html found ()

### ISSUE THREAD

**Issue by johansmitsnl** [intellij-rust/intellij-rust#6732]
## Environment

* **IntelliJ Rust plugin version:** 0.3.139.3615-203
* **Rust toolchain version:** 1.49.0
* **IDE name and version:** RubyMine 2020.3.1
* **Operating system:** Ubuntu 20.04 x64

## Problem description
When wrapping the html in the match it gives this markup error:

![Schermafdruk van 2021-01-27 09-33-57](https://user-images.githubusercontent.com/12995632/105964742-e80b4600-6082-11eb-9918-20de090f0f36.png)

## Steps to reproduce

Wrap a view in a match case like this:

```rust
match self.props.kind {
            ViewKind::Name => {
                html! {
                    <>
                         <>{ &account.name }</>
                    </>
                }
            }
            ViewKind::Avatar => {
                html! {
                    <>
                        <img src=&account.avatar_url class="avatar img-fluid avatar-xs mr-3" alt=&account.name />
                    </>
                }
            }
        }
```

**Comment by Waridley** [comment]
With `maud`'s `html!` macro, the error goes away if you change the braces to square or parens. Is the same true for `yew`?

**Comment by ctron** [comment]
For me using `html![]` or `html!()` seems to work, but fails in a different way. It then detects the code as unreachable.

**Comment by bit-void** [comment]
This is still happening on nightly. Switching to `[]` fixes the issue for me, but why does it fail with `html!{}`?

**Comment by ctron** [comment]
We switched to Yew 0.19 a while back. That actually "solved" the issue.

**Comment by bit-void** [comment]
FWIW, I'm on nightly yew and nightly rust plugin and this is still broken.

**Comment by hamza1311** [comment]
> nightly yew

Yew does not have a nightly release. You may want to clarify what you're referring to

**Comment by bit-void** [comment]
@hamza1311 the master branch: `yew = { git = "https://github.com/yewstack/yew/" }`

**Comment by dima74** [comment]
@johansmitsnl @Waridley @ctron @bit-void do you still experience this problem? If yes, could you please provide minimal example? I can't reproduce it with `0.4.176.4762-221-nightly` and `yew = { git = "https://github.com/yewstack/yew/" }`:

<img src="https://user-images.githubusercontent.com/6505554/183114624-aff824ad-5d28-4363-8e92-dd2598a61b91.png" width="50%">

**Comment by ctron** [comment]
No, I haven't seen this in a while. IIRC there was a significant improvement when migrating from Yew 0.17 to 0.19 (not sure about 0.18). I guess this improved something, which made it work better with the Rust plugin. It is ok to close from my point of view.

**Comment by johansmitsnl** [comment]
@dima74 I have not used yew projects lately so I can't give feedback on it. But @ctron does not experiencing it you can close it if you like.

**Comment by YthanZhang** [comment]
Hi, I just encountered this issue, the project is very minimal, only containing `Cargo.toml` and `src/main.rs`

main.rs
```rust
use yew::prelude::*;

fn main() {
   yew::start_app::<App>();
}


#[function_component(App)]
pub fn app() -> Html {
    html! {
        <div>
            <h2 class={"heading"}>{"Hello, World!"}</h2>
        </div>
    }
}
```

Cargo.toml
```toml
[package]
name = "yew-test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
yew = "0.19.3"
```

## Environment

+ OS windows 10
+ CLion 2022.2 Build 222.3345.126
+ Rust plugin 0.4.175.4772-222
+ Rust toolchain 1.62.1


##

Interestingly if I replace the `html! {}` with `html! ()` or `html! []` the error disappears.

I tried setting toolchain to nightly, but that didn't work.

**Comment by dima74** [comment]
@YthanZhang could you please try to enable [`org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` experimental features](https://github.com/intellij-rust/intellij-rust/issues/6908)

**Comment by YthanZhang** [comment]
@dima74 Enabling `org.rust.cargo.evaluate.build.scripts` fixed the issue. Enabling `org.rust.macros.proc` didn't seem to affect this issue.

**Comment by johansmitsnl** [comment]
With this project I can reproduce it easily: https://github.com/johan-smits/sycamore

`org.rust.cargo.evaluate.build.scripts` fixes the issue the component warning to be lowercase. In the `#[component]` they have added this ignore message.
![Schermafdruk van 2022-09-15 19-12-36](https://user-images.githubusercontent.com/12995632/190468413-be584c89-22e6-4a42-bbe3-6379d2405953.png)


`org.rust.macros.proc` fixes the issue that i says the wrong return type the initial error from this issue:
![Schermafdruk van 2022-09-15 19-05-45](https://user-images.githubusercontent.com/12995632/190468523-11e06c0e-be2b-48a5-a1f7-4ccf03121c8d.png)

With both options enabled it is now happy :+1: 

Hope this helps.

**Comment by bradvoth** [comment]
Both experimental features enabled and I still see this issue.  However changing to [] from {} resolves.  The totality of the app is as follows:
```use yew::prelude::*;

struct App {}

impl Component for App {
    type Message = ();
    type Properties = ();

    fn create(_ctx: &Context<Self>) -> Self {
        todo!()
    }

    fn view(&self, _ctx: &Context<Self>) -> Html {
        html! {}
    }
}

fn main() {
    yew::start_app::<App>();
}
```

**Comment by dima74** [comment]
> Both experimental features enabled and I still see this issue. However changing to [] from {} resolves

@bradvoth This is very strange. Is the error with `{}` consistently reproducible? Please try to [invalidate caches](https://www.jetbrains.com/help/idea/invalidate-caches.html). Also could you please attach your [`idea.log`](https://intellij-support.jetbrains.com/hc/en-us/articles/207241085-Locating-IDE-log-files) and specify what version of `yew` are you using
